modules/aws/ssm.go

```diff
@@ -68,3 +73,127 @@ func NewSsmClientE(t testing.TestingT, region string) (*ssm.SSM, error) {
+
+		if len(resp.Entities) != 1 {
+			return "", fmt.Errorf("%s is not in the SSM inventory", instanceID)
+		}
+
+		return "", nil
+	})
+
+	return err
+}
+
+// WaitForSsmInstance waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstance(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) {
+	err := WaitForSsmInstanceE(t, awsRegion, instanceID, timeout)
+	require.NoError(t, err)
+}
+
+// CheckSsmCommand checks that you can run the given command on the given instance through AWS SSM. Returns stdout and stderr.
+func CheckSsmCommand(t testing.TestingT, awsRegion, instanceID, command string, timeout time.Duration) (string, string) {
+	stdout, stderr, err := CheckSsmCommandE(t, awsRegion, instanceID, command, timeout)
+	if err != nil {
+		t.Fatalf("failed to execute '%s' on %s (%v):]\n  stdout: %#v\n  stderr: %#v", command, instanceID, err, stdout, stderr)
+	}
+	return stdout, stderr
+}
+
+type result struct {
+	Stdout string
+	Stderr string
+}
+
+// CheckSsmCommandE checks that you can run the given command on the given instance through AWS SSM. Returns the stdout, stderr and an error if one occurs.
+func CheckSsmCommandE(t testing.TestingT, awsRegion, instanceID, command string, timeout time.Duration) (string, string, error) {
+	logger.Logf(t, "Running command '%s' on EC2 instance with ID '%s'", command, instanceID)
+
+	timeBetweenRetries := 2 * time.Second
+	maxRetries := int(timeout.Seconds() / timeBetweenRetries.Seconds())
+
+	// Now that we know the instance in the SSM inventory, we can send the command
+	client, err := NewSsmClientE(t, awsRegion)
+	if err != nil {
+		return "", "", err
+	}
+	comment := "Terratest SSM"
+	documentName := "AWS-RunShellScript"
+	req, resp := client.SendCommandRequest(&ssm.SendCommandInput{
+		Comment:      &comment,
+		DocumentName: &documentName,
+		InstanceIds:  []*string{&instanceID},
+		Parameters: map[string][]*string{
+			"commands": []*string{&command},
+		},
+	})
+	if err := req.Send(); err != nil {
+		return "", "", err
+	}
+
+	// Wait for the result
+	description := "Waiting for the result of the command"
+	retryableErrors := map[string]string{
+		"InvocationDoesNotExist": "InvocationDoesNotExist",
+		"bad status: Pending":    "bad status: Pending",
+		"bad status: InProgress": "bad status: InProgress",
+		"bad status: Delayed":    "bad status: Delayed",
+	}
+	var stdout, stderr string
+	_, err = retry.DoWithRetryableErrorsE(t, description, retryableErrors, maxRetries, timeBetweenRetries, func() (string, error) {
+		req, resp := client.GetCommandInvocationRequest(&ssm.GetCommandInvocationInput{
+			CommandId:  resp.Command.CommandId,
+			InstanceId: &instanceID,
+		})
+		if err := req.Send(); err != nil {
+			return "", err
+		}
+
+		// Remove the SSM prefix from stderr
+		stderr = aws.StringValue(resp.StandardErrorContent)
+
+		stdout = aws.StringValue(resp.StandardOutputContent)
+
+		status := aws.StringValue(resp.Status)
+
+		if status == "Success" {
```

Comment:
NIT: Is this defined as a constant somewhere in the AWS Go SDK? Same with `"Failed"` below?